### ISSUE TITLE
iOS devices crashes on version 5.0.0 and 5.0.1

### ISSUE THREAD

**Issue by Faaatman** [AgoraIO/Agora-Flutter-SDK#634]
**Describe the bug**
iOS devices on both iOS 15 and iOS 14 started crashing when upgraded to version 5.0.0 and 5.0.1
It is happening in release mode only and I can't reproduce it. But it is happening a lot.

**Screenshots**
<img width="1073" alt="Screen Shot 2022-03-24 at 10 50 46 AM" src="https://user-images.githubusercontent.com/30783585/159868595-83d42be3-7988-41fb-85b0-2b4363eb41ee.png">

**Smartphone:**
 - OS: iOS
 - Version 15, 14

**Additional context**
<details>
<summary>These are the logs from firebase crashylitcs</summary>

```Crashed: com.apple.main-thread
0  agora_rtc_engine               0x7b30 -[AgoraRtcChannelPlugin dealloc] + 36
1  agora_rtc_engine               0x87cc -[AgoraRtcEnginePlugin .cxx_destruct] + 52
2  libobjc.A.dylib                0x7d28 object_cxxDestructFromClass(objc_object*, objc_class*) + 116
3  libobjc.A.dylib                0x4b74 objc_destructInstance + 80
4  libobjc.A.dylib                0xe1c8 _objc_rootDealloc + 80
5  agora_rtc_engine               0x81e4 -[AgoraRtcEnginePlugin dealloc] + 108
6  libsystem_blocks.dylib         0x17e8 _Block_release + 200
7  Flutter                        0x142bc (Missing UUID be105c8b7c053a8ba51f83e291732357)
8  libsystem_blocks.dylib         0x17e8 _Block_release + 200
9  Flutter                        0x3a7b4 (Missing UUID be105c8b7c053a8ba51f83e291732357)
10 Flutter                        0x3a800 (Missing UUID be105c8b7c053a8ba51f83e291732357)
11 Flutter                        0x42ecc (Missing UUID be105c8b7c053a8ba51f83e291732357)
12 Flutter                        0x41874 (Missing UUID be105c8b7c053a8ba51f83e291732357)
13 Flutter                        0x3dd228 (Missing UUID be105c8b7c053a8ba51f83e291732357)
14 Flutter                        0x3dcbac (Missing UUID be105c8b7c053a8ba51f83e291732357)
15 Flutter                        0x13e00 (Missing UUID be105c8b7c053a8ba51f83e291732357)
16 Flutter                        0xff4c (Missing UUID be105c8b7c053a8ba51f83e291732357)
17 CoreFoundation                 0x2af88 __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 28
18 CoreFoundation                 0xc6c3c ___CFXRegistrationPost_block_invoke + 52
19 CoreFoundation                 0x99ff4 _CFXRegistrationPost + 456
20 CoreFoundation                 0x40df4 _CFXNotificationPost + 716
21 Foundation                     0x1af58 -[NSNotificationCenter postNotificationName:object:userInfo:] + 96
22 UIKitCore                      0xf5513c -[UIApplication _terminateWithStatus:] + 300
23 UIKitCore                      0x63c080 -[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:] + 128
24 UIKitCore                      0x830fc8 -[_UISceneLifecycleMultiplexer forceExitWithTransitionContext:scene:] + 224
25 UIKitCore                      0xf51538 -[UIApplication workspaceShouldExit:withTransitionContext:] + 212
26 FrontBoardServices             0x4f780 __63-[FBSWorkspaceScenesClient willTerminateWithTransitionContext:]_block_invoke_2 + 80
27 FrontBoardServices             0xa06c -[FBSWorkspace _calloutQueue_executeCalloutFromSource:withBlock:] + 240
28 FrontBoardServices             0x4f718 __63-[FBSWorkspaceScenesClient willTerminateWithTransitionContext:]_block_invoke + 132
29 libdispatch.dylib              0x4670 _dispatch_client_callout + 20
30 libdispatch.dylib              0x8128 _dispatch_block_invoke_direct + 264
31 FrontBoardServices             0xb9c4 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 48
32 FrontBoardServices             0xad8c -[FBSSerialQueue _targetQueue_performNextIfPossible] + 220
33 FrontBoardServices             0xf414 -[FBSSerialQueue _performNextFromRunLoopSource] + 28
34 CoreFoundation                 0xbb0d0 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28
35 CoreFoundation                 0xcbd90 __CFRunLoopDoSource0 + 208
36 CoreFoundation                 0x6098 __CFRunLoopDoSources0 + 268
37 CoreFoundation                 0xb8a4 __CFRunLoopRun + 820
38 CoreFoundation                 0x1f468 CFRunLoopRunSpecific + 600
39 GraphicsServices               0x138c GSEventRunModal + 164
40 UIKitCore                      0x5195d0 -[UIApplication _run] + 1100
41 UIKitCore                      0x297f74 UIApplicationMain + 364
42 Runner                         0x7860 main + 7 (main.m:7)
43 ???                            0x1016ddaa4 (Missing)```

</details>

**Comment by LichKing-2234** [comment]
you should use .dsym file to get which line crashed

**Comment by Faaatman** [comment]
> you should use .dsym file to get which line crashed

I did upload them from local archives but apparently, the UUIDs don't match. And I couldn't download them from AppStore connect.
I downgraded to version **4.0.7** and the crashes stopped.

**Comment by LichKing-2234** [comment]
Fine, We will do more testing for it.

**Comment by Faaatman** [comment]
<img width="1105" alt="Screen Shot 2022-03-27 at 12 01 24 PM" src="https://user-images.githubusercontent.com/30783585/160274482-3e29cd78-2c4e-49c6-8cb5-447761758ded.png">
This banner is showing in firebase crashlytics. Unfortunately, I am facing that issue where I can't download the .dsym files.

**Comment by Dararii** [comment]
We experienced the same crash with version 5.0.1 in iOS 15 and 14. It should be in line 30.

<img width="1126" alt="Screen Shot 2022-03-27 at 22 36 37" src="https://user-images.githubusercontent.com/7982335/160289070-21ffc0d8-9b05-4785-b8cb-70bb715b9fbb.png">

**Comment by hschk** [comment]
We have the exact same error since version 5.0.0.
The app crashes as soon as I close the app (I mean really closing it, not just in background)

**Comment by littleGnAl** [comment]
Thanks for the report, this will be fixed in the next release.

**Comment by iamEtornam** [comment]
I'm facing the same issue. Hope this gets fixed soon

**Comment by kazy-dev** [comment]
https://github.com/AgoraIO/Agora-Flutter-SDK/issues/634#issuecomment-1080624744
> We have the exact same error since version 5.0.0. The app crashes as soon as I close the app (I mean really closing it, not just in background)
I don't know if it's the same here.
The following console log was displayed in Xcode when destroying a screen using agora in 5.0.0.
(I actually read the AgoraRtcEngine instance at the point where the screen is being destroyed, dispose, etc.)
```
[2022-04-05 11:43:34.284] [0] [debug] /tmp/jenkins/IRIS-SDK/rtc/cxx/src/iris_rtc_engine.cc:104 CallApi api_type 6 params {}
[2022-04-05 11:43:34.285] [0] [debug] /tmp/jenkins/IRIS-SDK/rtc/cxx/src/iris_rtc_engine.cc:114 CallApi ret 0 result 
[2022-04-05 11:43:34.286] [0] [error] /tmp/jenkins/IRIS-SDK/rtc/cxx/src/internal/rtc_engine_event_handler.cc:53 onError err 18 msg nullptr
[2022-04-05 11:43:34.288] [0] [debug] /tmp/jenkins/IRIS-SDK/rtc/cxx/src/iris_rtc_engine.cc:104 CallApi api_type 1 params {"sync":true}
[2022-04-05 11:43:34.315] [0] [debug] /tmp/jenkins/IRIS-SDK/rtc/cxx/src/iris_rtc_engine.cc:114 CallApi ret 0 result 
````

**Comment by tarek360** [comment]
> Thanks for the report, this will be fixed in the next release.

@littleGnAl  any plan for the next release, please?

**Comment by littleGnAl** [comment]
This issue is fixed, pls try upgrade agora_rtc_engine: 5.1.0
https://pub.dev/packages/agora_rtc_engine
